expo/eas-cli
# [24/n][eas-cli] Start to move graphql into a context

# Checklist


# Why

There is one more major piece that can be moved to context: the graphql client. By moving this to the context:
- Global shared state is reduced (or eliminated). Previously the current user and graphql client were stored globally, and whatever was in the current session (access token or session token) was injected into the graphql request on every call no matter whether the call needed to be authenticated or not.
- We can more accurately determine which commands need to be logged-in. Previously this was done with a member variable and by moving this into the context we can determine this with 100% accuracy (since the tokens that signify being logged-in are only used in graphql).
- We can further isolate the internals of the authentication mechanisms from commands that don't need to access them. This provides the more complex authentication API only to commands that need it (login, logout) and the rest of the other commands now don't need access to anything but the graphql client from the context.

# How

This first PR simulates a move of the graphql client to a context member. These changes were done procedurally by solely using the typechecker.

Here's how this PR was generated:
1. Rename `graphqlClient` to `legacyGraphqlClient`. Inject it in via method arguments all the way up and down the call tree in places that require it.
2. Make `LoggedInContextField` (replacement for `ActorContextField`) that requires login and exposes the actor and `legacyGraphqlClient` (a graphql client that will be replaced with a logged-in graphql client (a client configured to use the access token / session secret belonging the actor) in a follow-up PR in this stack).
3. Use `yarn tsc` to make all replacements and pipe the graphql field all the way from the context down to the leaves (usually the queries or mutations). 

This allows for a few things:
- Better testability. Since these are now injected, we can mock them (using ts-mockito) without needing to mock them globally (using jest mock).

# Test Plan

- Run all changed commands, ensure their graphql features are working.
- Run all tests.
